### PR TITLE
[ci] Migrate macOS build jobs to 10.15 Catalina

### PR DESCRIPTION
**Pull request by pjcollins**
Our macOS builds which run on Redmond hardware have been migrated to
macOS Catalina.  Two new agent pools have been created for this, and the
existing Untrusted agent pool is still in use.

The Untrusted pool has both Mojave and Catalina machines.

We can make sure to always pick a Catalina machine by setting a
[demand][0] for the relevant system [capability][1] that any given
[machine may have][2].

The .NET and Xcode versions that we provison have been bumped to match
what is availabe on the [default image][3] for these new machines.

I ran into a couple of build issues while working with this new pool
which required some changes in the src/monodroid build.   Both of these
errors were due to the removal of the xamarin-windeps homebrew recipe:

    [31/31] : && /usr/local/bin/x86_64-w64-mingw32-g++  -static-libgcc -static-libgcc  -fno-strict-aliasing -ffunction-sections -funswitch-loops -finline-limit=300 -fvisibility=hidden -fstack-protector-strong -Wa,-noexecstack -fPIC -g -O2 -fno-rtti -fno-exceptions -Wall -Wconversion -Wdeprecated -Wduplicated-branches -Wduplicated-cond -Werror=format-security -Werror=return-type -Wextra -Wformat -Wformat-security -Wmisleading-indentation -Wnull-dereference -Wpointer-arith -Wshadow -Wsign-compare -Wuninitialized  -DWINDOWS -DNTDDI_VERSION=NTDDI_VISTA -D_WIN32_WINNT=_WIN32_WINNT_VISTA -fomit-frame-pointer  -Wl,--no-undefined -Wl,--export-all-symbols -shared -o /Users/builder/azdo/_work/1/s/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64/libmono-android.debug.dll -Wl,--out-implib,/Users/builder/azdo/_work/1/s/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64/libmono-android.debug.dll.a -Wl,--major-image-version,0,--minor-image-version,0 CMakeFiles/mono-android.debug.dir/jni/designer-assemblies.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop-gc-bridge-mono.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop-jvm.cc.obj CMakeFiles/mono-android.debug.dir/jni/new_delete.cc.obj CMakeFiles/mono-android.debug.dir/jni/android-system.cc.obj CMakeFiles/mono-android.debug.dir/jni/basic-android-system.cc.obj CMakeFiles/mono-android.debug.dir/jni/basic-utilities.cc.obj CMakeFiles/mono-android.debug.dir/jni/cpu-arch-detect.cc.obj CMakeFiles/mono-android.debug.dir/jni/debug.cc.obj CMakeFiles/mono-android.debug.dir/jni/debug-constants.cc.obj CMakeFiles/mono-android.debug.dir/jni/embedded-assemblies.cc.obj CMakeFiles/mono-android.debug.dir/jni/embedded-assemblies-zip.cc.obj CMakeFiles/mono-android.debug.dir/jni/globals.cc.obj CMakeFiles/mono-android.debug.dir/jni/logger.cc.obj CMakeFiles/mono-android.debug.dir/jni/monodroid-glue.cc.obj CMakeFiles/mono-android.debug.dir/jni/xa-internal-api.cc.obj CMakeFiles/mono-android.debug.dir/jni/osbridge.cc.obj CMakeFiles/mono-android.debug.dir/jni/shared-constants.cc.obj CMakeFiles/mono-android.debug.dir/jni/timezones.cc.obj CMakeFiles/mono-android.debug.dir/jni/util.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop-mono.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop-util.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop-dlfcn.cc.obj -L/Users/builder/azdo/_work/1/s/xamarin-android/src/monodroid/../../bin/BuildRelease/mingw-deps/x86_64/lib  -L/Users/builder/azdo/_work/1/s/xamarin-android/src/monodroid/../../bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64 -lmonosgen-2.0.dll /usr/local/opt/mingw-zlib/usr/x86_64-w64-mingw32/lib/libz.a -lmman -static -pthread -dynamic -lmincore -lmswsock -lwsock32 -lshlwapi -lpsapi -lwinmm /Users/builder/azdo/_work/1/s/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64/libxamarin-app.dll.a -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 && :
    FAILED: /Users/builder/azdo/_work/1/s/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64/libmono-android.debug.dll /Users/builder/azdo/_work/1/s/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64/libmono-android.debug.dll.a 
    x86_64-w64-mingw32-g++ : error : /usr/local/opt/mingw-zlib/usr/x86_64-w64-mingw32/lib/libz.a: No such file or directory [/Users/builder/azdo/_work/1/s/xamarin-android/src/monodroid/monodroid.csproj]
    ninja: build stopped: subcommand failed.

    [31/31] : && /usr/local/bin/x86_64-w64-mingw32-g++  -static-libgcc  -fno-strict-aliasing -ffunction-sections -funswitch-loops -finline-limit=300 -fvisibility=hidden -fstack-protector-strong -Wa,-noexecstack -fPIC -g -O2 -fno-rtti -fno-exceptions -Wall -Wconversion -Wdeprecated -Wduplicated-branches -Wduplicated-cond -Werror=format-security -Werror=return-type -Wextra -Wformat -Wformat-security -Wmisleading-indentation -Wnull-dereference -Wpointer-arith -Wshadow -Wsign-compare -Wuninitialized  -DWINDOWS -DNTDDI_VERSION=NTDDI_VISTA -D_WIN32_WINNT=_WIN32_WINNT_VISTA -fomit-frame-pointer  -Wl,--no-undefined -Wl,--export-all-symbols -shared -o /Users/builder/azdo/_work/1/s/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64/libmono-android.debug.dll -Wl,--out-implib,/Users/builder/azdo/_work/1/s/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64/libmono-android.debug.dll.a -Wl,--major-image-version,0,--minor-image-version,0 CMakeFiles/mono-android.debug.dir/jni/designer-assemblies.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop-gc-bridge-mono.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop-jvm.cc.obj CMakeFiles/mono-android.debug.dir/jni/new_delete.cc.obj CMakeFiles/mono-android.debug.dir/jni/android-system.cc.obj CMakeFiles/mono-android.debug.dir/jni/basic-android-system.cc.obj CMakeFiles/mono-android.debug.dir/jni/basic-utilities.cc.obj CMakeFiles/mono-android.debug.dir/jni/cpu-arch-detect.cc.obj CMakeFiles/mono-android.debug.dir/jni/debug.cc.obj CMakeFiles/mono-android.debug.dir/jni/debug-constants.cc.obj CMakeFiles/mono-android.debug.dir/jni/embedded-assemblies.cc.obj CMakeFiles/mono-android.debug.dir/jni/embedded-assemblies-zip.cc.obj CMakeFiles/mono-android.debug.dir/jni/globals.cc.obj CMakeFiles/mono-android.debug.dir/jni/logger.cc.obj CMakeFiles/mono-android.debug.dir/jni/monodroid-glue.cc.obj CMakeFiles/mono-android.debug.dir/jni/xa-internal-api.cc.obj CMakeFiles/mono-android.debug.dir/jni/osbridge.cc.obj CMakeFiles/mono-android.debug.dir/jni/shared-constants.cc.obj CMakeFiles/mono-android.debug.dir/jni/timezones.cc.obj CMakeFiles/mono-android.debug.dir/jni/util.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop-mono.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop-util.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop-dlfcn.cc.obj -L/Users/builder/azdo/_work/1/s/xamarin-android/src/monodroid/../../bin/BuildRelease/mingw-deps/x86_64/lib  -L/Users/builder/azdo/_work/1/s/xamarin-android/src/monodroid/../../bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64 -lmonosgen-2.0 -lz -lmman -static -pthread -dynamic -lmincore -lmswsock -lwsock32 -lshlwapi -lpsapi -lwinmm /Users/builder/azdo/_work/1/s/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64/libxamarin-app.dll.a -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 && :
    FAILED: /Users/builder/azdo/_work/1/s/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64/libmono-android.debug.dll /Users/builder/azdo/_work/1/s/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64/libmono-android.debug.dll.a 
    /usr/local/Cellar/mingw-w64/8.0.0_2/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: cannot find -lmonosgen-2.0
    /usr/local/Cellar/mingw-w64/8.0.0_2/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: cannot find -lz
    collect2 : error : ld returned 1 exit status [/Users/builder/azdo/_work/1/s/xamarin-android/src/monodroid/monodroid.csproj]
    ninja: build stopped: subcommand failed.

[0]: https://docs.microsoft.com/en-us/azure/devops/pipelines/process/demands?view=azure-devops&tabs=yaml#manually-entered-demands
[1]: https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/agents?view=azure-devops&tabs=browser#capabilities
[2]: https://devdiv.visualstudio.com/DevDiv/_settings/agentqueues?agentId=688191&queueId=2206&view=capabilities
[3]: https://github.com/xamarinhq/bot-deploy-package-generation/pull/3

### CODE REVIEW

**Review by radekdoulik — APPROVED** [review]
Nice to see green  PR builds again :-)

**Review by grendello — APPROVED** [review]

File: src/monodroid/monodroid.props [hunk 1]
```diff
@@ -14,8 +14,8 @@
     <_CmakeMxeCommonFlags>$(_CmakeCommonFlags) $(_CommonDefines) -DENABLE_NDK=OFF</_CmakeMxeCommonFlags>
 
     <_CmakeMxeCommonFlags Condition=" '$(_MingwDependenciesRootDirectory)' != '' ">$(_CmakeMxeCommonFlags) -DMINGW_DEPENDENCIES_ROOT_DIR=$(_MingwDependenciesRootDirectory)</_CmakeMxeCommonFlags>
-    <_CmakeMxeCommonFlags32>$(_CmakeMxeCommonFlags) -DCMAKE_TOOLCHAIN_FILE=$(_CmakeMxeToolchainBasePath)-32.cmake -DMINGW_TARGET_32=1 -DMINGW_ZLIB_LIBRARY_PATH=&quot;$(MingwZlibRootDirectory32)\lib\$(MingwZlibLibraryName)&quot;</_CmakeMxeCommonFlags32>
```
Comment (jonpryor) [inline src/monodroid/monodroid.props:17]: What's the rationale/commit message content for this & related changes? It doesn't appear to be directly related to Catalina, is it?

Comment (pjcollins) [reply src/monodroid/monodroid.props:17]: The `-DMINGW_ZLIB_LIBRARY_PATH` parameter was causing issues now that we've removed the xamarin-windeps homebrew recipe.  I've updated the PR description with the two build errors I hit while working with this new pool initially.

**Comment by jonpryor** [comment]
Squash and merge:

Summary

```
[ci] Migrate macOS build jobs to 10.15 Catalina (#5589)
```

Body:

```markdown
Our macOS builds which run on Redmond hardware have been migrated to
macOS Catalina.  Two new agent pools have been created for this, and
the existing `VSEng-Xamarin-RedmondMac-Android-Untrusted` agent pool
is still in use.

The `VSEng-Xamarin-RedmondMac-Android-Untrusted` pool has both Mojave
and Catalina machines.

We can make sure to always pick a Catalina machine by setting a
[demand][0] for the relevant system [capability][1] that any given
[machine may have][2].

The .NET and Xcode versions that we provision have been bumped to
match what is available on the [default image][3] for these machines:

  * .NET Core 3.1.405
  * Xcode 12.4

I ran into a couple of build issues while working with this new pool
which required some changes to the `src/monodroid` build.   Both of
these errors were due to the removal of the xamarin-windeps homebrew
recipe (c1679796):

	[31/31] : && /usr/local/bin/x86_64-w64-mingw32-g++  -static-libgcc -static-libgcc  -fno-strict-aliasing -ffunction-sections -funswitch-loops -finline-limit=300 -fvisibility=hidden -fstack-protector-strong -Wa,-noexecstack -fPIC -g -O2 -fno-rtti -fno-exceptions -Wall -Wconversion -Wdeprecated -Wduplicated-branches -Wduplicated-cond -Werror=format-security -Werror=return-type -Wextra -Wformat -Wformat-security -Wmisleading-indentation -Wnull-dereference -Wpointer-arith -Wshadow -Wsign-compare -Wuninitialized  -DWINDOWS -DNTDDI_VERSION=NTDDI_VISTA -D_WIN32_WINNT=_WIN32_WINNT_VISTA -fomit-frame-pointer  -Wl,--no-undefined -Wl,--export-all-symbols -shared -o /Users/builder/azdo/_work/1/s/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64/libmono-android.debug.dll -Wl,--out-implib,/Users/builder/azdo/_work/1/s/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64/libmono-android.debug.dll.a -Wl,--major-image-version,0,--minor-image-version,0 CMakeFiles/mono-android.debug.dir/jni/designer-assemblies.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop-gc-bridge-mono.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop-jvm.cc.obj CMakeFiles/mono-android.debug.dir/jni/new_delete.cc.obj CMakeFiles/mono-android.debug.dir/jni/android-system.cc.obj CMakeFiles/mono-android.debug.dir/jni/basic-android-system.cc.obj CMakeFiles/mono-android.debug.dir/jni/basic-utilities.cc.obj CMakeFiles/mono-android.debug.dir/jni/cpu-arch-detect.cc.obj CMakeFiles/mono-android.debug.dir/jni/debug.cc.obj CMakeFiles/mono-android.debug.dir/jni/debug-constants.cc.obj CMakeFiles/mono-android.debug.dir/jni/embedded-assemblies.cc.obj CMakeFiles/mono-android.debug.dir/jni/embedded-assemblies-zip.cc.obj CMakeFiles/mono-android.debug.dir/jni/globals.cc.obj CMakeFiles/mono-android.debug.dir/jni/logger.cc.obj CMakeFiles/mono-android.debug.dir/jni/monodroid-glue.cc.obj CMakeFiles/mono-android.debug.dir/jni/xa-internal-api.cc.obj CMakeFiles/mono-android.debug.dir/jni/osbridge.cc.obj CMakeFiles/mono-android.debug.dir/jni/shared-constants.cc.obj CMakeFiles/mono-android.debug.dir/jni/timezones.cc.obj CMakeFiles/mono-android.debug.dir/jni/util.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop-mono.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop-util.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop-dlfcn.cc.obj -L/Users/builder/azdo/_work/1/s/xamarin-android/src/monodroid/../../bin/BuildRelease/mingw-deps/x86_64/lib  -L/Users/builder/azdo/_work/1/s/xamarin-android/src/monodroid/../../bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64 -lmonosgen-2.0.dll /usr/local/opt/mingw-zlib/usr/x86_64-w64-mingw32/lib/libz.a -lmman -static -pthread -dynamic -lmincore -lmswsock -lwsock32 -lshlwapi -lpsapi -lwinmm /Users/builder/azdo/_work/1/s/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64/libxamarin-app.dll.a -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 && :
	FAILED: /Users/builder/azdo/_work/1/s/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64/libmono-android.debug.dll /Users/builder/azdo/_work/1/s/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64/libmono-android.debug.dll.a 
	x86_64-w64-mingw32-g++ : error : /usr/local/opt/mingw-zlib/usr/x86_64-w64-mingw32/lib/libz.a: No such file or directory [/Users/builder/azdo/_work/1/s/xamarin-android/src/monodroid/monodroid.csproj]
	ninja: build stopped: subcommand failed.


	[31/31] : && /usr/local/bin/x86_64-w64-mingw32-g++  -static-libgcc  -fno-strict-aliasing -ffunction-sections -funswitch-loops -finline-limit=300 -fvisibility=hidden -fstack-protector-strong -Wa,-noexecstack -fPIC -g -O2 -fno-rtti -fno-exceptions -Wall -Wconversion -Wdeprecated -Wduplicated-branches -Wduplicated-cond -Werror=format-security -Werror=return-type -Wextra -Wformat -Wformat-security -Wmisleading-indentation -Wnull-dereference -Wpointer-arith -Wshadow -Wsign-compare -Wuninitialized  -DWINDOWS -DNTDDI_VERSION=NTDDI_VISTA -D_WIN32_WINNT=_WIN32_WINNT_VISTA -fomit-frame-pointer  -Wl,--no-undefined -Wl,--export-all-symbols -shared -o /Users/builder/azdo/_work/1/s/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64/libmono-android.debug.dll -Wl,--out-implib,/Users/builder/azdo/_work/1/s/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64/libmono-android.debug.dll.a -Wl,--major-image-version,0,--minor-image-version,0 CMakeFiles/mono-android.debug.dir/jni/designer-assemblies.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop-gc-bridge-mono.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop-jvm.cc.obj CMakeFiles/mono-android.debug.dir/jni/new_delete.cc.obj CMakeFiles/mono-android.debug.dir/jni/android-system.cc.obj CMakeFiles/mono-android.debug.dir/jni/basic-android-system.cc.obj CMakeFiles/mono-android.debug.dir/jni/basic-utilities.cc.obj CMakeFiles/mono-android.debug.dir/jni/cpu-arch-detect.cc.obj CMakeFiles/mono-android.debug.dir/jni/debug.cc.obj CMakeFiles/mono-android.debug.dir/jni/debug-constants.cc.obj CMakeFiles/mono-android.debug.dir/jni/embedded-assemblies.cc.obj CMakeFiles/mono-android.debug.dir/jni/embedded-assemblies-zip.cc.obj CMakeFiles/mono-android.debug.dir/jni/globals.cc.obj CMakeFiles/mono-android.debug.dir/jni/logger.cc.obj CMakeFiles/mono-android.debug.dir/jni/monodroid-glue.cc.obj CMakeFiles/mono-android.debug.dir/jni/xa-internal-api.cc.obj CMakeFiles/mono-android.debug.dir/jni/osbridge.cc.obj CMakeFiles/mono-android.debug.dir/jni/shared-constants.cc.obj CMakeFiles/mono-android.debug.dir/jni/timezones.cc.obj CMakeFiles/mono-android.debug.dir/jni/util.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop-mono.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop-util.cc.obj CMakeFiles/mono-android.debug.dir/Users/builder/azdo/_work/1/s/xamarin-android/external/Java.Interop/src/java-interop/java-interop-dlfcn.cc.obj -L/Users/builder/azdo/_work/1/s/xamarin-android/src/monodroid/../../bin/BuildRelease/mingw-deps/x86_64/lib  -L/Users/builder/azdo/_work/1/s/xamarin-android/src/monodroid/../../bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64 -lmonosgen-2.0 -lz -lmman -static -pthread -dynamic -lmincore -lmswsock -lwsock32 -lshlwapi -lpsapi -lwinmm /Users/builder/azdo/_work/1/s/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64/libxamarin-app.dll.a -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 && :
	FAILED: /Users/builder/azdo/_work/1/s/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64/libmono-android.debug.dll /Users/builder/azdo/_work/1/s/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-mxe-Win64/libmono-android.debug.dll.a 
	/usr/local/Cellar/mingw-w64/8.0.0_2/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: cannot find -lmonosgen-2.0
	/usr/local/Cellar/mingw-w64/8.0.0_2/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: cannot find -lz
	collect2 : error : ld returned 1 exit status [/Users/builder/azdo/_work/1/s/xamarin-android/src/monodroid/monodroid.csproj]
	ninja: build stopped: subcommand failed.

Fix these errors by updating `$(_CmakeMxeCommonFlags32)` to remove
`-DMINGW_ZLIB_LIBRARY_PATH`, as that path no longer exists.

[0]: https://docs.microsoft.com/en-us/azure/devops/pipelines/process/demands?view=azure-devops&tabs=yaml#manually-entered-demands
[1]: https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/agents?view=azure-devops&tabs=browser#capabilities
[2]: https://devdiv.visualstudio.com/DevDiv/_settings/agentqueues?agentId=688191&queueId=2206&view=capabilities
[3]: https://github.com/xamarinhq/bot-deploy-package-generation/pull/3
```

**Pull request merged**